odoo/models.py

```diff
@@ -2437,27 +2438,57 @@ def _inherits_join_add(self, current_model, parent_model_name, query):
         return parent_alias
 
     @api.model
-    def _inherits_join_calc(self, alias, fname, query):
+    def _inherits_join_calc(self, alias, fname, query, allow_related=False):
```

Comment:
I don't think the boolean flag is actually necessary.  The method could handle related fields in all cases.